src/pkgcheck/checks/profiles.py

```diff
@@ -146,28 +192,34 @@ class ProfilesCheck(Check):
     _source = sources.ProfilesRepoSource
     required_addons = (addons.UseAddon, addons.KeywordsAddon)
     known_results = frozenset(
-        [
+        {
             UnknownProfilePackage,
             UnmatchedProfilePackageUnmask,
             UnknownProfilePackageUse,
             UnknownProfileUse,
             UnknownProfilePackageKeywords,
             UnknownProfileUseExpand,
+            UnknownProfileUseExpandValue,
+            ProfileMissingImplicitExpandValues,
+            UnknownProfileArch,
             ProfileWarning,
             ProfileError,
-        ]
+        }
     )
 
     # mapping between known files and verification methods
     known_files = {}
 
-    def __init__(self, *args, use_addon, keywords_addon):
+    def __init__(self, *args, use_addon: addons.UseAddon, keywords_addon: addons.KeywordsAddon):
         super().__init__(*args)
         repo = self.options.target_repo
         self.keywords = keywords_addon
         self.search_repo = self.options.search_repo
         self.profiles_dir = repo.config.profiles_base
-        self.use_expand_groups = frozenset(x.upper() for x in repo.config.use_expand_desc)
+        self.use_expand_groups = {
```

Comment:
Any thoughts on moving the concept of use_expand groups into pkgcore itself?  We've got multiple spots that are now hardcoding how to split them.  Seems like something where an object representation might be of value.

Reply:
I will create a shared impl in pkgcore, but for now I don't won't to be blocked until pkgcore release, so for now duplication of code, until I introduce the new shared code.

Reply:
No worries.  My comment is more an observation I did something similar in portage_conf for use_expand, so it's probably time one of us cuts a class to hold the rules for it. ;)